test/test_prototype_models.py

```diff
@@ -65,10 +76,9 @@ def test_get_weight(model_fn, name, weight):
     + TM.get_models_from_module(models.video),
 )
 def test_naming_conventions(model_fn):
-    model_name = model_fn.__name__
-    module = _get_parent_module(model_fn)
-    weights_name = "_QuantizedWeights" if module.__name__.split(".")[-1] == "quantization" else "_Weights"
-    assert model_name in set(x.replace(weights_name, "").lower() for x in module.__dict__ if x.endswith(weights_name))
+    weights_enum = _get_model_weights(model_fn)
+    assert weights_enum is not None
+    assert len(weights_enum) == 0 or hasattr(weights_enum, "default")
```

Comment:
It's not obvious at first sight what this line is testing?

Reply:
The first thing I test here is that every model builder has a WeightsEnum named like `capitalized(model_builder) + "_Weights"`. The capitalization used on the class is not the same as on the builder, so I check on the lower-case string. Effectively I am ensuring that all of the prototypes use the same naming conventions. 

The second test checks that all WeightsEnums have defined a default weight. The only ones allowed not to define any is those enums that don't have any pretrained weights (we got a couple of cases).

The fact that you have questions here is a sign that the unit-test is not readable and easily understood (key rule for unit-tests). As I don't plan to move these tests in the main area, these were checks to simplify my development so if you are OK, I don't what to spent time improving them.

Reply:
> The only ones allowed not to define any is those enums that don't have any pretrained weights (we got a couple of cases).

This was the bit that I was missing, the rest is clear - no need to change anything

> The first thing I test here is that every model builder has a WeightsEnum named like capitalized(model_builder) + "_Weights". The capitalization used on the class is not the same as on the builder, so I check on the lower-case string. Effectively I am ensuring that all of the prototypes use the same naming conventions

That's a good test to have IMO, but I think it was removed lol